functions/tab.xfce4_terminal.fish

```diff
@@ -0,0 +1,3 @@
+function tab.xfce4_terminal -a cdto cmd
+  xfce4-terminal --tab --working-directory "$cdto" -e "fish -c '$cmd; exec fish'"
```

Comment:
It gives a proper interactive shell when the $cmd finishes or is stopped. 

The outstanding issues are:
  (a) $cmd is not a child process of the shell -- so while ctl-z will suspend a running process, fg won't resume it. 
  (b) the resulting shell's history would be better with $cmd in it, rather than 'tab $cmd', or nothing; I've encountered both. Achieving this requires modifying history, which appears to be a bit tricky. Still working on it.

There's a good bash-centric discussion of the above at http://unix.stackexchange.com/questions/75902/how-to-run-xfce-terminal-with-different-commands-per-tab-and-keep-using-the-tabs with some interesting ideas, but I haven't yet found any good way to translate into fish.